meta/classes/build_yocto.bbclass

```diff
@@ -0,0 +1,106 @@
+# N.B. this is guaranteed by the layer priority (see layer.conf
+# BBFILE_PRIORITY_xt-core = "5") that this append will
+# run after base_do_unpack and BEFORE any other unpacks
+# such as proprietary code unpack in other layers etc.
+# so, the repo will get updated properly before other
+# related recipes
+build_yocto_repo_sync() {
```

Comment:
I guess this should be moved out build_yocto.bbclass to some generic override for repo fetcher (also with separate commit). You need the same functionality for any stuff you will be getting using .repo, i.e. android sources.